### ISSUE TITLE
Autoreconf fails

### ISSUE THREAD

**Issue by Smaloki** [ShabbyX/vktut#7]
Just downloaded the latest version of the project, now trying to run "autoreconf -i" throws a warning and an error:

`aclocal: warning: couldn't open directory 'm4': No such file or directory`
`configure.ac:21: error: possibly undefined macro: AC_MSG_ERROR`
`If this token and others are legitimate, please use m4_pattern_allow.`
`See the Autoconf documentation.`
`autoreconf: /usr/bin/autoconf failed with exit status: 1`

**Comment by ShabbyX** [comment]
Strange indeed. My first suggestion is to edit configure.ac and change this line:

```
  AC_MSG_ERROR([C99 support is required]))
```

to this:

```
  [AC_MSG_ERROR([C99 support is required])])
```

(which adds some quoting). If that doesn't solve the issue, try the answers [here](http://stackoverflow.com/q/8811381/912144). Are you by any chance on a 32-bit OS?

**Comment by Smaloki** [comment]
I'm using a 64-bit system.

Adding the brackets changed nothing; I did, however, notice that executing the command a second time leads to a different result (for whatever reason). Apparently the script actually works after the initial failure. It gives this output:

`aclocal: warning: couldn't open directory 'm4': No such file or directory`
`configure.ac:15: installing 'bin/ar-lib'`
`configure.ac:13: installing 'bin/compile'`
`configure.ac:5: installing 'bin/config.guess'`
`configure.ac:5: installing 'bin/config.sub'`
`configure.ac:7: installing 'bin/install-sh'`
`configure.ac:7: installing 'bin/missing'`
`Makefile.am: installing 'bin/depcomp'`

When trying to run configure afterwards, it results in a new error:

`checking for clock_gettime in -lrt... yes`
`../configure: line 4884: syntax error near unexpected token 'newline'`
`../configure: line 4884: 'AX_PTHREAD('`

The first time I installed the project (2 days ago), the script managed to proceed until the part where it looks for the Vulkan header - this time, it stops earlier, when checking for pthread. So I went and checked what had changed in the last two days - and I noticed that some lines involving pthread had been changed. I reverted these changes in my version of the script (using AC_CHECK_HEADERS and AC_CHECK_LIBS to check for pthread), but it still displayed the same error... which makes no sense whatsoever, since the line "AX_PTHREAD(" (which is quoted in the error description) was commented out (and yes, I made sure that my changes to the script had been saved; and I restarted my terminal and tried again).

I'll get back to checking the other fix suggestions in the thread you linked. I already made sure that pkg-config is installed, but there are a few more answers to that post.

**Comment by ShabbyX** [comment]
Alright, regarding `AX_PTHREAD`, it was a suggestion made to me to improve the portability of the code. In case you didn't notice, I have added a small comment in the `README.md` file that says you would need to install the `autoconf-archive` package.

**Comment by ShabbyX** [comment]
Let me know if that fixed your issue.

**Comment by Smaloki** [comment]
I Installed autoconf-archive and tried again, still the same syntax error.

Edit: scratch that, ax_pthread is now working. I forgot to run autoreconf again after installing the autoconf-archive package, so of course my configure script was still broken...
It's working now; except it again - or rather, still - can't find the Vulkan headers.

**Comment by ShabbyX** [comment]
Hmmm... Can you give the output of these commands (omit personal information if necessary)?

```
$ echo $VULKAN_SDK
$ test -f "$VULKAN_SDK/include/vulkan/vulkan.h" && echo yes || echo no
```

Also, in `configure.ac`, which of these do you see?

```
  CPPFLAGS="$CPPFLAGS -I$VULKAN_SDK/include"
```

or

```
  CPPFLAGS="$CPPFLAGS -I'$VULKAN_SDK/include'"
```

(Note the single quote)

**Comment by Smaloki** [comment]
The two terminal commands:
`~$ echo $VULKAN_SDK`
`/home/salim/VulkanSDK/1.0.17.0/x86_64`
`~$ test -f "$VULKAN_SDK/include/vulkan/vulkan.h" && echo yes || no`
`yes`

And my `configure.ac` contains
`CPPFLAGS="$CPPFLAGS -I$VULKAN_SDK/include"`

I'll quickly try it _with_ the quotes.

Edit: same error, still won't find the header.

**Comment by ShabbyX** [comment]
Yeah, the quotes were an error, I thought maybe you have the old version.

So your files are where they are expected. Can you post the output of the `configure` script? I am expecting it to say it found the header, and if there is an error, it would be when it tries to use it. If it says it can't even see the header, that would be kind of contradictory to the output of the commands I mentioned above, but we'll see.

**Comment by Smaloki** [comment]
Got it working!

Out of desperation, I went back to the closed issue thread and tried your first recommendation (which I never got to try because you updated the script so quickly).
Passing the flags directly to the script when executing it seems to have worked:

`../configure CPPFLAGS="-I$VULKAN_SDK/include/" LDFLAGS="-L/$VULKAN_SDK/lib"`
... script does its thing ...
`checking vulkan/vulkan.h usability... yes`
`checking vulkan/vulkan.h presence... yes`
`checking for vulkan/vulkan.h... yes`
`checking for vkCreateInstance in -lvulkan... yes`
... checking for Vulkan header was successful ...
`checking that generated files are newer than configure... done`
`configure: creating ./config.status`
`config.status: creating Makefile`
`config.status: creating config.h`
`config.status: executing depfiles commands`
... and we're done!

I'm still not sure why the flags defined in the script itself were ignored, but at least I got it working. I think I'll try and play around with the script a little, to see if I can find the difference.

Anyway... thanks again for helping! Especially for being so patient with me...

Compared to my struggle trying to get this set up, actually learning Vulkan should be a breeze :D
